rest-team.go

```diff
@@ -0,0 +1,291 @@
+package sdk
+
+/*
+   Copyright 2016 Alexander I.Grafov <grafov@gmail.com>
+   Copyright 2016-2019 The Grafana SDK authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+	   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+
+   ॐ तारे तुत्तारे तुरे स्व
+*/
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"strconv"
+)
+
+// SearchTeams search teams with optional parameters.
+// Reflects GET /api/teams/search API call.
+func (r *Client) SearchTeams(ctx context.Context, params ...SearchTeamParams) (PageTeams, error) {
+	var (
+		raw       []byte
+		pageTeams PageTeams
+		code      int
+		err       error
+		requestParams = make(url.Values)
+	)
+
+	for _, p := range params {
+		p(requestParams)
+	}
+
+	if raw, code, err = r.get(ctx, "api/teams/search", requestParams); err != nil {
+		return pageTeams, err
+	}
+	if code != 200 {
+		return pageTeams, fmt.Errorf("HTTP error %d: returns %s", code, raw)
+	}
+	dec := json.NewDecoder(bytes.NewReader(raw))
+	dec.UseNumber()
+	if err := dec.Decode(&pageTeams); err != nil {
+		return pageTeams, fmt.Errorf("unmarshal teams: %s\n%s", err, raw)
+	}
+	return pageTeams, err
+}
+
+func (r *Client) GetTeamByName(ctx context.Context, name string) (*Team, error) {
```

Comment:
```suggestion
func (r *Client) GetTeamByName(ctx context.Context, name string) (Team, error) {
```
I'd suggest returning `Team` directly instead of pointer to it just like the other methods for consistency :smile: